lib/creator/yeoman/utils/entry.js

```diff
@@ -0,0 +1,50 @@
+const Input = require('webpack-addons').Input;
+
+module.exports = (self, answer) => {
+	return new Promise( (resolve) => {
```

Comment:
It's already async, so not sure what you're talking about, but here's how you could do it (Note how the promise is always returned):

```js
const InputValidate = require('webpack-addons').InputValidate;
const validate = require('./validate');

module.exports = (self, answer) => {
	let webpackEntryPoint;

	let entryIdentifiers;
	let entryProperties;

	let ret;

	if(answer['entryType'] == 'Multiple') {
		ret = self.prompt([
			InputValidate(
				'multipleEntries',
				'Write the name you want for your modules seperated by comma',
				validate
			)
		]).then( (multipleEntriesAnswer) => {
			webpackEntryPoint = {};
			entryIdentifiers = multipleEntriesAnswer['multipleEntries'].split(',');
			return self.prompt([
				InputValidate(
					'objectProperties',
					'Write the location of those modules seperated by comma',
					validate
				)
			]).then( (objectPropAnswer) => {
				entryProperties = objectPropAnswer['objectProperties'].split(',');
				for(let k = 0; k < entryIdentifiers.length; k++) {
					if(entryProperties[k].charAt(0) == '(' ||  entryProperties[k].charAt(0) == '[') {
						webpackEntryPoint[entryIdentifiers[k]] = entryProperties[k];
					} else {
						webpackEntryPoint[entryIdentifiers[k]] = `'${entryProperties[k]}.js'`;
					}
				}
				return webpackEntryPoint;
			});
		});
	}
	else {
		ret = self.prompt([
			InputValidate(
				'singularEntry',
				'Write the location of module you would like to use',
				validate
			)
		]).then( (singularAnswer) => `'${singularAnswer['singularEntry']}.js'`);
	}

	return ret;
};
```